### PR TITLE
feat(plan): the /magic:plan skill — brainstorm, spec artifact, GitHub ticket creation

### LINKED ISSUE #198: feat(plan): the /magic:plan skill — brainstorm, spec artifact, GitHub ticket creation
Part of #170. Depends on #196 (statuses + `specPath`) and #197 (`languages.ticket` + the `plan` block).

This is the skill itself: the flow, the spec artifact, the agent metadata, the registration of the 8th skill, and creation against **GitHub**. Jira creation is #199 — same epic, later merge, because it is the product's first Jira write and it should not gate the rest. The tracker abstraction (`references/trackers.md`) is written **here**, with GitHub as its first implementation, so #199 fills in a branch rather than reshaping the skill.

### 0. Bootstrap (aligned with the other skills)
Read `~/.config/magic-slash/config.json`, resolve `languages.discussion`, read `~/.config/magic-slash/profile.md` to adapt tone and technical depth.

### 1. Capture the idea
From the argument (`argument-hint: <idea>`), or via `AskUserQuestion` when the skill is invoked bare.

### 2. Ask for the target repository — the key step
`AskUserQuestion` listing the repositories from config. Smart pre-selection: match the idea against each repo's `keywords`, and surface the one matching the current `pwd` first. The user always keeps the final say.

This step is what makes the brainstorm relevant: without the repository, we can neither explore the code, nor detect the tracker, nor check for duplicates.

**Metadata write #1 happens here**, and the spec file is created — both sections below.

### 3. Contextual brainstorm
- Explore the target repository's codebase (`Explore` agent, to keep the main context clean)
- Read the target repo's `CLAUDE.md` when present — conventions, stack, structure
- Search existing issues to surface duplicates and related work (when `plan.duplicateCheck` is on)

Findings land in the spec as they are established.

### 4. Framing dialogue
`AskUserQuestion` on genuinely ambiguous points only: scope, target users, success criteria, explicit non-goals. Never ask what the code or the config already answers.

Each resolved answer is written into the spec — the spec is the running record of what has been decided.

### 5. Sizing
Classify the idea with an explicit, documented heuristic:
- **Single story** — one coherent deliverable, one reasonable PR
- **Epic + N stories** — several independently mergeable deliverables, or several surfaces touched (skills / desktop / webapp)

The breakdown must produce stories that are **independently shippable and reviewable**, not artificial horizontal slices ("the backend", "the frontend"). Eagerness follows `plan.splitting`.

### 6. Review the spec, then approve
The spec is now complete. The user reviews it — in the desktop panel (#171) or simply in their editor — and approves the proposed ticket structure before anything is created. The user can adjust the breakdown, merge or split stories; edits go back into the spec.

**This step is not configurable.** A setting that lets tickets be created without review is a setting that eventually floods someone's backlog.

**Metadata write #2 happens here** — the title firms up once the structure is agreed.

### 7. Ticket creation
Tracker resolved from `plan.tracker`, falling back to the repo's `issues.githubIssuesUrl` / `issues.jiraUrl`, and asking only when still ambiguous. In this story only the GitHub branch is implemented; selecting Jira reports that it is not available yet rather than failing at write time.

- **GitHub**: `mcp__github__issue_write`. Real hierarchy via `mcp__github__sub_issue_write` — the epic becomes a parent issue with its stories as sub-issues.

Ticket bodies are generated **from the spec**, not from conversation memory, and written in `languages.ticket`.

**Metadata write #3 happens here** — the ticket ID now exists. The created ticket URLs are appended to the spec.

### 8. Chain
Offer `/magic:start <first story>`.

## Spec artifact

`/magic:start` already writes `.magic/design-brief.md` and treats it as a first-class object — sub-agents read it, the plan fills a section from it, the critic grades against it. `/magic:plan` follows the same pattern one step earlier in the cycle.

The spec is not a byproduct. It is **the reviewable object**: what the user approves in step 6, and what ticket bodies are generated from in step 7. Generating tickets from the spec rather than from conversation memory is what makes "review before creation" mean something — what you read is what gets created.

It is also useful with no UI at all: a markdown file in the project, openable in any editor. The desktop rendering in #171 is a convenience layer, not a prerequisite.

### Location and naming

- [ ] Write to `.magic/spec-<YYYYMMDD-HHmm>-<slug>.md` at the **target repository root**. `/magic:plan` creates no worktree, so the per-worktree placement `design-brief.md` uses does not apply.
- [ ] **The timestamp prefix is not cosmetic.** The slug derives from the idea, so planning the same idea twice on one repository yields the same filename — and #194 keys its cloud row on a hash of that path, so the second session would silently overwrite the first. The timestamp also gives `.magic/` a natural chronological sort.
- [ ] Exclude `.magic/` from git via `.git/info/exclude`, reusing the exact logic in `skills/magic-start/references/design-context.md:108-119` — including the `grep -qxF` guard and the trailing-newline guard that avoid the `node_modules.magic/` false positive
- [ ] Do **not** delete a pre-existing spec the way `/magic:start` deletes a stale design brief. A previous spec on the same repo is a different plan, not a stale artifact of this one.

### Content

- [ ] Written progressively, not in one shot at the end — the file is the live record of the session, and #171 renders it as it fills
- [ ] Structure: the idea as stated, framing decisions and their rationale, codebase findings, related/duplicate tickets found, the sizing verdict with its justification, the proposed ticket breakdown with acceptance criteria per story
- [ ] Acceptance criteria follow `plan.acceptanceCriteria`
- [ ] Written in `languages.ticket` — it is the source of the ticket bodies, so it must be in the same language as them
- [ ] After creation, append the created ticket IDs and URLs, so the spec remains a usable record of what came out of the session

## Agent metadata

`/magic:start` does not just prepare a branch — it names the agent in the desktop app. `/magic:plan` must do the same, otherwise a ten-minute brainstorm sits in the UI as an unnamed agent with no repository.

The endpoints and the two new fields exist as of #196. Skills hit them with `curl` guarded by `$MAGIC_SLASH_PORT` and `$MAGIC_SLASH_TERMINAL_ID`, every value through `jq -sRr @uri`, every call ending in `|| true`. Reference implementation: `skills/magic-start/SKILL.md` step 2.5.2 and the `/repositories` call at line 368.

| When | Fields | Why then |
| --- | --- | --- |
| **Step 2** — repository chosen | `/repositories` with the repo path, plus `title` = short form of the idea, `status=planning`, `specPath` = absolute path of the spec | Earliest moment we know anything useful. The agent stops being anonymous before the long exploration starts, and the desktop learns which file to watch. |
| **Step 6** — structure approved | `title` refined to the agreed epic/story wording | The idea's phrasing and the final ticket title usually differ; the user has just approved the latter. |
| **Step 7** — tickets created | `ticketId`, `title` = `TICKET-ID: Title` (same convention as `/magic:start`), `description`, `status=planned` | The ticket ID only exists now. |

- [ ] `specPath` is sent **at file creation time**, before the brainstorm starts — the consumers tolerate the file not existing yet, but not a path that arrives ten minutes late
- [ ] **`branchName` / `baseBranch` are deliberately never set.** `/magic:plan` creates no branch — it must not claim one. State this explicitly in the `SKILL.md` so nobody "fixes" it later.
- [ ] **`description` is never passed raw** — same caution as `/magic:start` step 2.5.1: ticket text gets URL-encoded into a query string
- [ ] On an epic breakdown, the agent carries the **epic** as its `ticketId` — that is what this agent produced; starting a story is a separate concern and `/magic:start <story>` drives its own agent. Worth confirming against how the sidebar groups agents.

### The skill
- [ ] `skills/magic-plan/SKILL.md` — frontmatter (`name: magic:plan`, `argument-hint`, `allowed-tools` including `mcp__github__*`), then steps 0 through 8
- [ ] `skills/magic-plan/references/messages.md` — bilingual `MSG_*` messages, starting from `.claude/skills/magic-plan/references/messages.md`
- [ ] `skills/magic-plan/references/trackers.md` — the tracker abstraction: detection, creation, parent/child hierarchy, required fields. GitHub implemented here, Jira slot left explicit for #199. This file is what makes adding Linear and others cheap later
- [ ] `skills/magic-plan/references/sizing.md` — story vs epic heuristic, breakdown rules, acceptance-criteria templates, and how `plan.splitting` shifts the threshold
- [ ] `skills/magic-plan/references/spec-template.md` — the spec's structure, so every session produces a comparable document

### Removing the internal skill
- [ ] Delete `.claude/skills/magic-plan/` — the shipped skill is a superset (multi-repo, multi-tracker, epic breakdown, spec artifact), and we dogfood our own product on this repo rather than maintaining two implementations
- [ ] Check no internal doc still points at it

### Registering the 8th skill
Adding a skill is not just creating its folder — the list is duplicated in several places:
- [ ] `desktop/src/main/skills-updater.ts:6` — `SKILLS` array
- [ ] `desktop/src/main/setup/status.ts:17` — `SKILLS` array (installation detection)
- [ ] `webapp/lib/skills.ts` — `TRACKED_SKILLS` (telemetry and stat tiles; note the "seven tiles" comment to update)
- [ ] `desktop/src/renderer/pages/QuickLaunch/index.tsx` — command entry + i18n key
- [ ] `desktop/src/renderer/pages/Dashboard/SkillStats.tsx`
- [ ] `webapp/lib/i18n/marketing/{en,fr}.ts` and `doc-{en,fr}.ts`
- [ ] `webapp/components/site/documentation/DocContent.tsx`
- [ ] `webapp/components/site/home/AppMockup.tsx` and `mockupAnimation.ts`
- [ ] `install/uninstall.sh`
- [ ] `commitlint.config.js` — add `plan` to `scope-enum`
- [ ] `README.md`, `CLAUDE.md` — "7 skills" → "8 skills" (README lines 73, 269, 436; CLAUDE.md lines 3, 9) and add it to the cycle list
- [ ] `CHANGELOG.md`
- [ ] **Add a test asserting `skills/*/` and the two `SKILLS` arrays stay aligned.** Ten hand-maintained lists is how the 9th skill gets half-registered; the test costs a few lines and this is the moment to write it.

### Evals
- [ ] `skills/evals/eval_set.json` — queries targeting `magic:plan` in EN and FR ("I have an idea", "j'ai une idée", "we should add", "brainstorm", "create a ticket for…")
- [ ] Add **negative**, discriminating cases: `/magic:plan` must not capture `/magic:start` phrasing. The boundary is thin — "I want to work on X" (start, the ticket exists) vs "what if we did X" (plan, nothing exists). This is the riskiest part of the skill.
- [ ] Run the evals and confirm no triggering regression on the 7 existing skills

## Acceptance criteria

- [ ] `/magic:plan "an idea"` explicitly asks for the target repository, with keyword-based pre-selection
- [ ] The agent shows the correct repository and a meaningful title in the desktop sidebar **from step 2**, before the brainstorm starts
- [ ] A `.magic/spec-<timestamp>-<slug>.md` is created at step 2 and fills in progressively through the session
- [ ] Planning the same idea twice on one repository produces two distinct files
- [ ] `.magic/` is git-excluded on the target repository, and the spec is never committed by `/magic:commit`
- [ ] Ticket bodies are generated from the spec, so what the user reviewed is what gets created
- [ ] Created ticket IDs and URLs are appended to the spec after creation
- [ ] The agent's title, description and ticket ID are set once tickets are created, following `/magic:start`'s `TICKET-ID: Title` convention
- [ ] The agent carries a `specPath` pointing at its own spec from step 2, and two planning agents on the same repository carry different ones
- [ ] The skill never sets `branchName` or `baseBranch`
- [ ] The brainstorm draws on the chosen repo's codebase and `CLAUDE.md`, not generalities
- [ ] Existing issues are searched and duplicates surfaced before anything is created (when `plan.duplicateCheck` is on)
- [ ] The skill classifies the idea as a single story or an epic + stories, with readable justification, and `plan.splitting` visibly shifts that threshold
- [ ] The proposed structure is approved by a human **before** anything is written to the tracker, with no way to opt out
- [ ] GitHub creation works, with a real epic/story hierarchy via sub-issues
- [ ] Choosing Jira reports that it is not available yet, before the brainstorm rather than after it
- [ ] Repository issue templates are honoured when `plan.useRepoTemplates` is on
- [ ] Spec and ticket content are written in `languages.ticket`, independently of `languages.discussion`
- [ ] The skill offers `/magic:start <ticket>` at the end
- [ ] The skill is bilingual and respects `languages.discussion` + `profile.md`
- [ ] Every `curl` to the status server is guarded and ends in `|| true` — the skill works with the desktop app closed
- [ ] Evals pass, with no triggering regression on the 7 existing skills
- [ ] `.claude/skills/magic-plan/` is deleted and this repository uses the shipped skill

## Risks

- 🟠 **Triggering overlap with `/magic:start`.** Both answer "I want to work on X". The discrimination has to be carried by the frontmatter `description` and covered by negative evals. This is the main product risk.
- 🟠 **Partial creation.** An epic created, then a failure on story 3 of 5, leaves the tracker half-built. Report explicitly what was created and what failed, with enough detail to resume. The spec makes recovery tractable — it holds everything needed to finish the job.
- 🟠 **`.magic/` exclusion is fiddly.** The `design-context.md` logic exists precisely because a naive `grep` matches `node_modules.magic/`. Reuse it verbatim rather than rewriting it.
- 🟢 **Skill list duplicated in 10+ places** — mitigated by the alignment test above.

## Impact analysis

- **Complexity**: L — the `SKILL.md` will land at the high end of the existing range (comparable skills run 350 to 820 lines), and registration touches a dozen files
- **Risk**: Medium
- **Breaking changes**: none. Purely additive skill; the other 7 are untouched.

## Architecture notes

The skill follows the established pattern exactly: `SKILL.md` as numbered steps + `references/` loaded on demand, bilingual messages isolated in `messages.md`, config and profile read in step 0, `AskUserQuestion` at genuine decision points, `Explore` agent delegation for heavy exploration, status-server `curl` calls for agent metadata, and a `.magic/` artifact as the durable output.

The only new building block is `references/trackers.md`. It deserves to exist here rather than later: it is the difference between "we support GitHub and Jira" and "we can support any tracker". Other skills will read that abstraction instead of rediscovering it when they in turn need to create tickets.

The spec deliberately mirrors `design-brief.md` — same directory, same git exclusion, same "a file other agents read" role. Matching that shape is what lets #171's desktop rendering be built once and later point at either file.

### PR DESCRIPTION
**Pull request by Xrequillart**
## Description

Adds `/magic:plan`, the eighth skill: it turns an idea into GitHub tickets. It brainstorms against the target repository's codebase, writes a reviewable spec at `.magic/spec-<slug>-<timestamp>.md`, and creates the epic and its stories only once a human has approved the structure.

Jira creation stays out of scope (#199), but the tracker abstraction ships here — `references/trackers.md` implements GitHub and leaves an explicit Jira slot, so #199 fills in a branch rather than reshaping the skill. Choosing Jira today stops the run at step 2, before the brainstorm, rather than failing at write time after ten minutes of exploration.

Everything the skill claims about the desktop side already exists on `main`: the `planning` / `planned` statuses, `specPath`, `languages.ticket` and the whole `plan` config block landed in #196 and #197.

## Related Issue

Closes #198

## Type of Change

- [x] New feature (non-breaking change that adds functionality)

## Changes Made

- **The skill** — `skills/magic-plan/SKILL.md` (463 lines, steps 0→9) plus five references: `messages.md` (bilingual `MSG_*`), `trackers.md` (the tracker abstraction), `sizing.md` (the single-story-vs-epic heuristic and how `plan.splitting` shifts it), `spec-template.md`, and `api.md` (duplicated verbatim, as the other skills do).
- **The internal skill is gone** — `.claude/skills/magic-plan/` is deleted; this repository now dogfoods the shipped skill. Nothing in the repo pointed at it.
- **Registered as the eighth skill** across the places that duplicate the list: the three desktop arrays (`skills-updater.ts`, `setup/status.ts`, `ipc/skills-handlers.ts`), QuickLaunch `COMMANDS` + `quickLaunch.cmd.plan` in both locales, both `TRACKED_SKILLS` rows, `HowSection.tsx`, the marketing and doc catalogues (EN/FR), `DocContent.tsx`, `install/uninstall.sh`, `commitlint.config.js`, `README.md`, `CLAUDE.md`, `package.json`, `CHANGELOG.md`, and the internal `magic-release` / `magic-audit` skills.
- **A new alignment test** — `desktop/src/main/skills-registry.test.ts` (25 tests) pins **eight** duplicated lists against `skills/*/SKILL.md` as the single source of truth, plus the three `grid-cols-8` Tailwind literals that encode an array length nothing else connects to them. The issue asked for the two `SKILLS` arrays; the other six drift just as quietly.
- **Evals** — 14 new cases (ids 45-58) in `skills/evals/eval_set.json`, purely additive.
- **Three tile rows widened** from `sm:grid-cols-7` to `sm:grid-cols-8`, otherwise the eighth tile wraps onto a second line on its own.

`/magic:plan` is deliberately **excluded** from the animated landing mockup (`AppMockup.tsx` / `mockupAnimation.ts`), for the reason the existing comment already gives for `/magic:continue`: it does not demonstrate linearly — its shape is a conversation that pauses on a human approval a non-interactive timeline cannot show. The issue's action plan listed those two files, so this is a documented deviation, not an omission. It also avoids shifting a 7-element `CONTEXT_BY_PHASE` array and a `switch (index)` with hard-coded cases, for no gain.

## Testing

- [ ] Tested locally with Claude Code
- [x] Ran linters (`npm run lint`)
- [ ] Tested installation script
- [ ] Tested affected slash commands
- Preview: https://magic-slash-9nw0j088o-xrequillarts-projects.vercel.app — this PR's actual deployed code; migrations, deletions and seeding still need local.

`npm test` → 82 files, 1255 tests passing. `npm run lint` → 0 errors (3 pre-existing `no-explicit-any` warnings in `Config/RepoPage.tsx`, untouched). `desktop` `tsc --noEmit` → no source errors.

Each of the four commits was checked out individually and its suite run: 1230 → 1230 → 1255 → 1255. The order matters — the alignment test would fail if it landed before the registration commit.

The two unchecked boxes are honest: the skill itself has not been executed end-to-end, and the eval set has not been run. See **Additional Notes**.

### Test Steps

1. **Break the alignment test on purpose.** Remove the `{ skill: 'magic-plan', label: '/magic:plan' }` line from `webapp/lib/skills.ts`, then run `npx vitest run desktop/src/main/skills-registry.test.ts`. Expect 3 failures naming the exact site. Restore the line: 25 tests pass again. Repeat with `sm:grid-cols-8` → `sm:grid-cols-7` in `webapp/components/SkillStats.tsx` — expect the grid test to fail for that file only.
2. **Run the suite and the linters.** `npm test` then `npm run lint` from the repo root. Expect 1255 tests passing and 0 lint errors.
3. **Check the launcher and the dashboard.** `npm run desktop`, open QuickLaunch and type `/magic:` — `/magic:plan` should appear first, described as "Turn an idea into tickets". On the Dashboard, the skill row should show 8 tiles on a single line at normal window width, not 7 plus 1.
4. **Check the docs page.** `cd webapp && npm i && npm run dev`, open the documentation → the `/magic:plan` block should be first in the skills section, with the `/magic:start` boundary stated in its Note. It has **no icon**, which is expected — see Additional Notes.
5. **Run the skill itself.** `/magic:plan "some idea"` should ask which repository to target before anything else, create `.magic/spec-<slug>-<timestamp>.md` **inside the repository you picked** (not the directory you invoked it from), fill that file as the session progresses, and create nothing in the tracker until you approve the structure. Running it twice on the same idea and repo must produce two distinct files.

## Screenshots

Not applicable — the only visual changes are one extra tile in an existing row and one extra block on the docs page.

## Checklist

- [x] My code follows the project's style guidelines
- [x] I have performed a self-review of my code
- [x] I have commented my code where necessary
- [x] I have updated the documentation accordingly
- [x] My changes generate no new warnings
- [x] I have added/updated tests if applicable
- [x] All linters pass locally
- [x] I have updated CHANGELOG.md (if applicable)

## Additional Notes

Three things a reviewer should weigh before merging:

- **The evals have not been run.** The issue asks for "evals pass, with no triggering regression on the 7 existing skills", and this PR cannot demonstrate it: there is no eval runner in the repo, and `skills/evals/results.json` still reports `run_date 2026-03-19` / `total_queries: 30` against a set that held 44 cases before this PR and 58 after. It was already stale, and generating results by hand would be fabricating them. **The riskiest case is id 55** — "we should add tests to the parser before we ship", expected `null` — because it contains the literal phrase "we should add" that the new frontmatter `description` advertises as a trigger. Worth running before merge.
- **No artwork.** `webapp/public/img/` holds exactly the seven `skill-*.png` and `design/` has no source, so `skills/magic-plan/image.png` and `skill-plan.png` are not in this PR. The docs block **omits** its `` rather than pointing at a 404; the CSS is `flex` + `gap`, so nothing collapses (the "create your own" block is iconless precedent). Add the icon back once the asset exists.
- **`CHANGELOG.md` gains an `## [Unreleased]` section**, a shape this file did not use before. `magic-release` inserts a version section generated from commits, so someone should confirm that flow consumes or clears `[Unreleased]` rather than leaving it stranded above the next version.

Two side effects that are expected but visible:

- Adding the entry to `setup/status.ts` makes the setup modal report a missing skill on every machine until the skill installs — and `skills-updater.ts` pulls from `main`, so not before this merges.
- A pre-existing bug found along the way, left untouched as out of scope: `skills/magic-pr/SKILL.md` has invalid YAML frontmatter (an unescaped `: ` inside `description`). Worth its own issue.

### CODE REVIEW

File: skills/magic-plan/SKILL.md [hunk 1]
```diff
@@ -0,0 +1,463 @@
...
+it fills, and an interrupted session must leave behind everything that was settled up to that point.
+
+### 2.5: Metadata — first write
+
+```bash
+[ -n "$MAGIC_SLASH_PORT" ] && [ -n "$MAGIC_SLASH_TERMINAL_ID" ] && curl -s "http://127.0.0.1:$MAGIC_SLASH_PORT/repositories?id=$MAGIC_SLASH_TERMINAL_ID&repos=$(echo -n '["{REPO_PATH}"]' | jq -sRr @uri)" > /dev/null 2>&1 || true
+```
+
+```bash
+[ -n "$MAGIC_SLASH_PORT" ] && [ -n "$MAGIC_SLASH_TERMINAL_ID" ] && curl -s "http://127.0.0.1:$MAGIC_SLASH_PORT/metadata?id=$MAGIC_SLASH_TERMINAL_ID&title=$(echo -n '{IDEA_SHORT}' | jq -sRr @uri)&status=planning&specPath=$(echo -n '{SPEC_ABS_PATH}' | jq -sRr @uri)" > /dev/null 2>&1 || true
```
[the remark this answers is not in the document]
Comment (Xrequillart) [reply skills/magic-plan/SKILL.md]: Fixed in 9897a13.

All three metadata blocks now bind the free-text values through a **quoted** heredoc (`<<'MS_EOF'`) and use them as `"$VAR"`, so nothing user-typed reaches the command line as a literal any more. `printf '%s'` replaces `echo -n` (portability, and a value starting with `-` would have been read as a flag), and the `repos` array is now built with `jq -nc --arg` instead of pasted between literal brackets.

Your diagnosis of the mechanism was right and worth being precise about: `jq -sRr @uri` encodes the value it *receives*, so it cannot protect a literal that already broke apart before jq ran. This matters more here than in the sibling skills because `/magic:plan`'s input is prose a human just typed, in a product used in French — `j'ai une idée d'export` is the normal case, not the adversarial one.

Verified in a real shell: `j'ai une idée d'export "compare" $HOME \`whoami\`` now survives intact and percent-encodes correctly, with no expansion.

The rule is documented in the `## Metadata contract` section as a correctness requirement, so a later tidy-up does not quietly reintroduce the single-quoted form.

File: skills/magic-plan/SKILL.md [hunk 2]
```diff
@@ -0,0 +1,463 @@
...
+### 3.2: Read the project's own conventions
+
+If the target repository has a `CLAUDE.md`, read it. It states the project's structure, stack and
+conventions, and it routinely answers questions that would otherwise be asked in Step 4 — which is
+exactly the kind of question that must not be asked.
+
+### 3.3: Check for duplicates
+
+When `plan.duplicateCheck` is `true` (the default), search the tracker for existing work before
+proposing anything new. Use `mcp__github__search_issues` with the strongest nouns from the idea,
```
[the remark this answers is not in the document]
Comment (Xrequillart) [reply skills/magic-plan/SKILL.md:288]: Fixed in 9897a13.

Confirmed and addressed: `MAGIC_SLASH_BASE_PERMISSIONS` in `desktop/src/main/hooks/claude-hooks-config.ts` is an explicit allowlist, and the `'mcp__github__'` entry further down belongs to `MAGIC_SLASH_PERMISSION_MARKERS`, which only *identifies* permissions Magic Slash owns for cleanup — it grants nothing. So the four tools would indeed have prompted mid-session.

Added exactly the four the skill actually references (verified by grepping `skills/magic-plan/`, nothing more): `search_issues`, `issue_write`, `sub_issue_write`, `get_me`.

The comment above them records why they are pre-approved rather than prompted: both moments land mid-run — the duplicate search before the brainstorm, creation right after the approval — so a prompt would interrupt precisely where the user is waiting. The two write tools are safe to grant for the same reason `update_issue` already is: `/magic:plan` cannot reach them until a human has approved the structure, which the skill states is not configurable.

`claude-hooks-config.test.ts` still passes (37 tests).

File: commitlint.config.js [hunk 3]
```diff
@@ -29,6 +29,7 @@ module.exports = {
         'review',
         'resolve',
         'done',
+        'plan',
```
[the remark this answers is not in the document]
Comment (Xrequillart) [reply commitlint.config.js:32]: Not applying this one — the premise does not hold, and here is the evidence.

The two scope lists in this repo are already in agreement. `CLAUDE.md:66` reads:

```
Scopes: `start`, `continue`, `commit`, `pr`, `review`, `resolve`, `done`, `plan`, `install`, `docs`, `deps`, `ci`, `readme`, `landing`, `slides`, `community`, `desktop`
```

Diffing that list against `scope-enum` yields no difference in either direction (the only extra token in the raw grep is `always`, which is commitlint's severity keyword, not a scope). `plan` is present on both sides — this PR updated them together, in the same commit.

Beyond that, adding `plan` is not incidental to this change: issue #198 mandates it explicitly ("`commitlint.config.js` — add `plan` to `scope-enum`"), because the skill ships as `skills/magic-plan/` and its commits need a scope. This PR's own four commits use `feat(plan)` / `test(plan)`, and `npx commitlint --from HEAD~4 --to HEAD` exits 0.

Happy to revisit if you can point at the specific closed list you have in mind that I have missed — but retaining the existing scope set, as suggested, would contradict the issue and reject the commits this PR is made of.

File: skills/magic-plan/SKILL.md [hunk 4]
```diff
@@ -0,0 +1,508 @@
...
+
+### 2.5: Metadata — first write
+
+Substitute the free text into a **quoted** heredoc, never into the command line — see
+`## Metadata contract` for why this shape is mandatory rather than stylistic.
+
+```bash
+IDEA_SHORT=$(cat <<'MS_EOF'
+{IDEA_SHORT}
+MS_EOF
```
[the remark this answers is not in the document]
Comment (Xrequillart) [reply skills/magic-plan/SKILL.md]: Fixed in 142a12f — and thank you, this was a real regression introduced by my own previous fix.

You are right that a fixed delimiter is not a fix, it just moves the boundary. Rather than pick a less guessable delimiter, I removed the class: the free-form values no longer travel through the command line at all. Each is written to a file under `.magic/` with the `Write` tool — no shell involved in placing it — and the shell reads it back:

```
$(jq -Rsr 'sub("\n$";"") | @uri' < .magic/.mp-title)
```

The command line now contains nothing but a fixed literal path, so there is no quoting scheme left to have a pathological input. `.magic/` already exists at that point (created in Step 2.4) and is git-excluded, so this costs no new directory and the files are deleted immediately after the call.

Rehearsed end to end against exactly the input you describe:

| input | transported as |
| --- | --- |
| `j'ai une idée d'export "x" $HOME \`id\`` | literal, correctly encoded |
| `line one` / `MS_EOF` / `rm -rf ~` (3 lines) | `'line one\nMS_EOF\nrm -rf ~'` — data, not syntax |

Testing also caught a bug in my first attempt at this: I had built the `repos` array with `@json` and then encoded again, which sends the server `"[\"…\"]"` — a JSON string — instead of `["…"]`. It is now `jq -c` plus a single encode, verified by decoding the result back and parsing it. The `## Metadata contract` section records both rules.

One correction on framing, for the record: the values here are short summaries the skill itself composes (the idea is capped at 30 chars, the description is explicitly summarised before this point), and they are the user's own text in the user's own shell — so this is not a privilege boundary and "command execution with the user's privileges" overstates it. The early-termination failure mode was real regardless, and worth closing properly rather than mitigating. Which is what this does.

File: commitlint.config.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (Xrequillart) [reply commitlint.config.js:32]: Appreciated — and yes, that rule sounds worth keeping.

To be concrete about what makes it checkable: in this repo the pair is `commitlint.config.js` (`scope-enum`) and `CLAUDE.md` (the `Scopes:` line under Conventions). A scope addition is only a divergence if it lands in one and not the other, so diffing the two lists is enough to tell a genuine drift from a synchronised change. Worth noting the raw token grep of the config also picks up `always` — commitlint's severity keyword, not a scope — which is the one false positive to filter.

Same shape as the eight-list alignment problem this PR adds a test for: hand-maintained duplicate lists drift silently, so the check belongs in CI rather than in a reviewer's head. `desktop/src/main/skills-registry.test.ts` now pins the skill lists that way; the scope pair would be a reasonable next candidate.
